url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/expert_system/Rhythmic_Block.java

```diff
@@ -0,0 +1,805 @@
+			rhythmicValues = new Rhythmic_Value [4];
+			rhythmicValues[0] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[3] = Rhythmic_Value.SEMI_QUAVER;
+			
+		}
+		else if(name == "SEMI_QUAVERS4_THIRD_AND_FORTH_OFF"){
+			rhythmicValues = new Rhythmic_Value [4];
+			rhythmicValues[0] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[2] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[3] = Rhythmic_Value.SEMI_QUAVER_REST;
+			
+		}
+		else if(name == "SEMI_QUAVERS4_FIRST_AND_THIRD_OFF"){
+			rhythmicValues = new Rhythmic_Value [4];
+			rhythmicValues[0] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[1] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[2] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[3] = Rhythmic_Value.SEMI_QUAVER;
+			
+		}
+		else if(name == "SEMI_QUAVERS4_SECOND_AND_FORTH_OFF"){
+			rhythmicValues = new Rhythmic_Value [4];
+			rhythmicValues[0] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.SEMI_QUAVER_REST;
+			
+		}
+		else if(name == "SEMI_QUAVERS8"){
+			rhythmicValues = new Rhythmic_Value [8];
+			rhythmicValues[0] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[2] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[4] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[5] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[6] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[7] = Rhythmic_Value.SEMI_QUAVER;
+		}
+		else if(name == "SEMI_QUAVERS8_FIRST_THIRD_FIFTH_AND_SEVENTH_OFF"){
+			rhythmicValues = new Rhythmic_Value [8];
+			rhythmicValues[0] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[1] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[2] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[3] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[4] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[5] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[6] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[7] = Rhythmic_Value.SEMI_QUAVER;
+		}
+		else if(name == "SEMI_QUAVERS8_SECOND_FORTH_SIXTH_AND_EIGTH_OFF"){
+			rhythmicValues = new Rhythmic_Value [8];
+			rhythmicValues[0] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[4] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[5] = Rhythmic_Value.SEMI_QUAVER_REST;
+			rhythmicValues[6] = Rhythmic_Value.SEMI_QUAVER;
+			rhythmicValues[7] = Rhythmic_Value.SEMI_QUAVER_REST;
+		}
+		else if(name == "SEMI_QUAVERS8_SECOND_THIRD_FIFTH_AND_SIXTH_OFF"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.